src/overlays/actors/ovl_Obj_Chan/z_obj_chan.c

```diff
@@ -31,49 +39,404 @@ const ActorInit Obj_Chan_InitVars = {
-static InitChainEntry D_80BBAB5C[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+void ObjChan_InitMain(ObjChan* this, GlobalContext* globalCtx); /* static */
+void ObjChan_InitPot(ObjChan* this, GlobalContext* globalCtx);  /* static */
+void func_80BBA488(ObjChan* this, GlobalContext* globalCtx);    /* static */
+void ObjChan_DrawPot(Actor* thisx, GlobalContext* globalCtx);   /* static */
+void func_80BBA930(ObjChan* this, GlobalContext* globalCtx);    /* static */
+
+static Vec3f sObjChanFlameSize[2] = { { 0.159999996424f, 0.109999999404f, 1.0f },
+                                      { 0.129999995232f, 0.0900000035763f, 1.0f } };
+static f32 sObjChanFlameYOffset[2] = { 1800, 1800 };
+static s32 sObjChanLoaded;
+
+void ObjChan_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s16 temp_t2;
+    ObjChan* this = THIS;
+
+    if (OBJCHAN_SUBTYPE(this) == OBJCHAN_SUBTYPE_MAIN) {
+        if (sObjChanLoaded) {
+            Actor_MarkForDeath(&this->actor);
+            return;
+        }
+        this->actor.room = -1;
+        sObjChanLoaded = true;
+    }
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sObjChanCylinderInit);
+    func_8013E3B8(&this->actor, this->cutscenes, ARRAY_COUNT(this->cutscenes));
+    switch (OBJCHAN_SUBTYPE(this)) {
+        case OBJCHAN_SUBTYPE_MAIN:
+            this->rotation = this->actor.shape.rot.y;
+            this->actor.shape.rot.y = 0;
+            ObjChan_InitMain(this, globalCtx);
+            break;
+        case OBJCHAN_SUBTYPE_POT:
+            this->actor.draw = ObjChan_DrawPot;
+            ObjChan_InitPot(this, globalCtx);
+            break;
+    }
+}
+
+void ObjChan_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjChan* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+u32 func_80BB9A1C(ObjChan* this, f32 arg1) {
+    f32 temp_f6;
+    f32 sp20;
+
+    sp20 = Math_SinS(this->unk1D4) * this->unk1D0;
+    temp_f6 = (Math_CosS(this->unk1D4) * 0.03834952f * this->unk1D0) + arg1;
+    if (temp_f6 != 0.0f) {
+        this->unk1D4 = (s16)(s32)(func_80086B30(sp20 * 0.03834952f, temp_f6) * 10430.378f);
+    } else if (sp20 >= 0.0f) {
+        this->unk1D4 = 0x4000;
+    } else {
+        this->unk1D4 = -0x4000;
+    }
+    if (Math_CosS(this->unk1D4) != 0.0f) {
+        this->unk1D0 = (temp_f6 / (Math_CosS(this->unk1D4) * 0.03834952f));
+    } else {
+        this->unk1D0 = sp20;
+    }
+    return 0;
+}
+
+void ObjChan_CalculatePotPosition(Vec3f* childPosOut, Vec3s* childRotOut, Vec3f* parentPos, Vec3s* parentRot,
+                                  s16 childAngle) {
+    Vec3f offset;
+
+    Matrix_RotateY(parentRot->y, false);
+    Matrix_InsertXRotation_s(parentRot->x, true);
+    Matrix_InsertZRotation_s(parentRot->z, true);
+    Matrix_RotateY(childAngle, true);
```

Comment:
```suggestion
    Matrix_RotateY(parentRot->y, MTXMODE_NEW);
    Matrix_InsertXRotation_s(parentRot->x, MTXMODE_APPLY);
    Matrix_InsertZRotation_s(parentRot->z, MTXMODE_APPLY);
    Matrix_RotateY(childAngle, MTXMODE_APPLY);
```
Here you can use an enum we have for the specific matrix values instead of true/false